src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1238,286 +1238,116 @@ void RequireDynamicallyAccessedMembers (ref ReflectionPatternContext reflectionC
 
 		void MarkTypeForDynamicallyAccessedMembers (ref ReflectionPatternContext reflectionContext, TypeDefinition typeDefinition, DynamicallyAccessedMemberTypes requiredMemberKinds)
 		{
-			if (requiredMemberKinds == DynamicallyAccessedMemberTypes.All) {
-				var sourceMethod = reflectionContext.SourceMethod;
-				reflectionContext.RecordRecognizedPattern (typeDefinition, () => _markStep.MarkEntireType (typeDefinition, includeBaseTypes: true, new DependencyInfo (DependencyKind.AccessedViaReflection, sourceMethod)));
-				return;
+			foreach (var member in DynamicallyAccessedMembersBinder.GetDynamicallyAccessedMembers (typeDefinition, requiredMemberKinds)) {
+				if (member == typeDefinition) {
```

Comment:
I would probably keep this a special case in the caller - and force everybody who calls the helper to handle All as a special case as well - it has somewhat special meaning compared to the other flags.